File: deploy/add_categories.py
#!/usr/bin/env python3
import json
import sys

# Read files
print("Reading files...")
with open('ingr_ingr_hub.json', 'r') as f:
    ingr_hub = json.load(f)

with open('network_data_hub.json', 'r') as f:
    network_data = json.load(f)

categories = network_data.get('categories', {})
matched = 0
unmatched = 0

# Add category to each node
print("Adding categories to nodes...")
for node in ingr_hub['nodes']:
    node_name = node.get('name')
    category_info = categories.get(node_name)
    
    if category_info:
        node['category'] = category_info.get('category', 'Unknown')
        node['categoryColor'] = category_info.get('color', '#95a5a6')
        matched += 1
    else:
        node['category'] = 'Unknown'
        node['categoryColor'] = '#95a5a6'
        unmatched += 1

# Write updated JSON
print(f"Writing updated file... Matched: {matched}, Unmatched: {unmatched}")
with open('ingr_ingr_hub.json', 'w') as f:
    json.dump(ingr_hub, f, indent=2)

print(f'✓ Categories added successfully! {matched} matched, {unmatched} unmatched')

File: deploy/add-categories.js
/**
 * Script to add category information to ingr_ingr_hub.json nodes
 * by matching with network_data_hub.json categories
 */

const fs = require('fs');
const path = require('path');

// Read both JSON files
const ingrHubPath = path.join(__dirname, 'ingr_ingr_hub.json');
const networkDataPath = path.join(__dirname, 'network_data_hub.json');

console.log('Reading files...');
const ingrHub = JSON.parse(fs.readFileSync(ingrHubPath, 'utf8'));
const networkData = JSON.parse(fs.readFileSync(networkDataPath, 'utf8'));

const categories = networkData.categories || {};
let matched = 0;
let unmatched = 0;

// Add category to each node
console.log('Adding categories to nodes...');
ingrHub.nodes.forEach(node => {
    const nodeName = node.name;
    const categoryInfo = categories[nodeName];
    
    if (categoryInfo) {
        node.category = categoryInfo.category;
        node.categoryColor = categoryInfo.color;
        matched++;
    } else {
        node.category = 'Unknown';
        node.categoryColor = '#95a5a6';
        unmatched++;
    }
});

// Write updated JSON
console.log(`Matched: ${matched}, Unmatched: ${unmatched}`);
console.log('Writing updated file...');
fs.writeFileSync(ingrHubPath, JSON.stringify(ingrHub, null, 2));
console.log('✓ Categories added successfully!');


